library/alloc/src/string.rs

```diff
@@ -3094,6 +3149,115 @@ impl fmt::Write for String {
     }
 }
 
+/// An iterator over the `[char]`s of a string.
+///
+/// This struct is created by the [`into_chars`] method on [`String`].
+/// See its documentation for more.
+///
+/// [`char`]: prim@char
+/// [`into_chars`]: String::into_chars
+#[cfg_attr(not(no_global_oom_handling), derive(Clone))]
+#[must_use = "iterators are lazy and do nothing unless consumed"]
+#[unstable(feature = "string_into_chars", issue = "133125")]
+pub struct IntoChars {
+    bytes: vec::IntoIter<u8>,
+}
+
+#[unstable(feature = "string_into_chars", issue = "133125")]
+impl fmt::Debug for IntoChars {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_tuple("IntoChars").field(&self.as_str()).finish()
+    }
+}
+
+impl IntoChars {
+    /// Views the underlying data as a subslice of the original data.
+    ///
+    /// This has the same lifetime as the original slice, and so the
+    /// iterator can continue to be used while this exists.
```

Comment:
This statement is false for `IntoChars`: the iterator cannot continue while the borrow is active. Otherwise the iterator could finish and free the string while the slice is still borrowed.

Reply:
I remove this comment.

I wonder if we can still have this method, since if there is an immutable reference to the iterator, we can't call `next` anyway.

Reply:
The method itself is fine, it's just the comment that was incorrect.

Reply:
Thanks for your reply. Then this patch is ready to merge.